### PR TITLE
Domains: Update subdomain connection instructions

### PR DESCRIPTION
**Pull request by leonardost**
### Changes proposed in this Pull Request

Up until now we were showing the same instructions to connect domains and subdomains, which could be confusing for customers who wanted to connect the latter, as they could mistakenly change their root domain's records or name servers. Cases like this were mentioned by HEs such as in p2MSmN-96M-p2.

This PR updates the instructions to connect subdomains to WPCOM, making them clearer and different from the instructions to connect root domains.

The following screenshots show the changes in the steps of both the suggested and advanced connection setup flows:

- Suggested setup

| Before | After |
| --- | --- |
| Markup on 2022-02-15 at 20:19:00 | Markup on 2022-02-25 at 16:36:07 |
| 154166711-629a09a2-fa8b-401c-ae10-d5cecd9cd7d9 | Screen Shot 2022-02-25 at 16 37 44 |

- Advanced setup

| Before | After |
| --- | --- |
| Markup on 2022-02-15 at 20:19:31 | Markup on 2022-02-25 at 16:41:33 |
| Screen Shot 2022-02-15 at 20 19 48 | Screen Shot 2022-02-24 at 21 23 39 |

### Testing instructions

- Build this branch locally or open the live Calypso link
- Connect a new subdomain to a site to access the connection setup, or access an already connected subdomain's setup via `/domains/mapping/<site_slug>/setup/<subdomain>`
- Ensure the instructions and recommended DNS record configurations were updated like in the screenshots above, both in the suggested and advanced flows
- Please also test connecting a root domain and ensure the instructions are the same as they were before

### CODE REVIEW

File: client/components/domains/connect-domain-step/connect-domain-step-advanced-start.jsx [hunk 1]
```diff
@@ -6,21 +6,34 @@ import CardHeading from 'calypso/components/card-heading';
...
 	const { __ } = useI18n();
-	const switchToSuggestedSetup = () => setPage( stepSlug.SUGGESTED_START );
+	const firstStep = isSubdomain( domain )
+		? stepSlug.SUBDOMAIN_SUGGESTED_START
+		: stepSlug.SUGGESTED_START;
+	const switchToSuggestedSetup = () => setPage( firstStep );
+
+	const message = isSubdomain( domain )
+		? __(
+				'This is the advanced way to connect your subdomain, using CNAME records. We advise using our <a>suggested setup</a> instead, with NS records.'
```
Comment (a8ci18n) [inline client/components/domains/connect-domain-step/connect-domain-step-advanced-start.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'This is the advanced way to connect your domain, using root A records & CNAME records. We advise using our <a>suggested setup</a> instead, with WordPress.com name servers.' )` *ES Score: 8*

File: client/components/domains/connect-domain-step/connect-domain-step-suggested-start.jsx [hunk 2]
```diff
@@ -9,35 +9,43 @@ import {
...
 	const { __ } = useI18n();
-	const switchToAdvancedSetup = () => setPage( stepSlug.ADVANCED_START );
+	const firstStep = isSubdomain( domain )
+		? stepSlug.SUBDOMAIN_ADVANCED_START
+		: stepSlug.ADVANCED_START;
+	const switchToAdvancedSetup = () => setPage( firstStep );
+
+	const message = isSubdomain( domain )
+		? __(
+				'This is the easiest way to connect your subdomain, using NS records. If needed you can also use our <a>advanced setup</a>, using CNAME records.'
```
Comment (a8ci18n) [inline client/components/domains/connect-domain-step/connect-domain-step-suggested-start.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'This is the easiest way to connect your domain, using name servers. If needed you can also use our <a>advanced setup</a>, using root A & CNAME records.' )` *ES Score: 7*

File: client/components/domains/connect-domain-step/page-definitions.js [hunk 3]
```diff
@@ -93,6 +95,87 @@ export const connectADomainStepsDefinition = {
...
+		},
+		component: ConnectDomainStepLogin,
+		next: stepSlug.SUBDOMAIN_ADVANCED_UPDATE,
+		prev: stepSlug.SUBDOMAIN_ADVANCED_START,
+	},
+	[ stepSlug.SUBDOMAIN_ADVANCED_UPDATE ]: {
+		mode: modeType.ADVANCED,
+		step: stepType.UPDATE_CNAME_RECORDS,
+		get name() {
+			return __( 'Update CNAME records' );
```
Comment (a8ci18n) [inline client/components/domains/connect-domain-step/page-definitions.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Update root A records & CNAME record' )` *ES Score: 8*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/subdomain-connection-instructions)

File: client/components/domains/connect-domain-step/connect-subdomain-step-suggested-records.jsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
+import { Button } from '@automattic/components';
```
Comment (leonardost) [inline client/components/domains/connect-domain-step/connect-subdomain-step-suggested-records.jsx:1]: This is basically a copy of `ConnectDomainStepAdvancedRecords` - I want to refactor this to not duplicate all this code, but also wanted to keep this PR simple for now 😅 let's refactor this in the future

**Review by gius80 — COMMENTED** [review]
Thank for this update! I tested and everything works as expected!

My only doubt is about the Suggested Setup, not all providers allow you to create NS of that type (GoDaddy does, AWS doesn't), this may perhaps ne confusing for some users (paradoxically I think the advanced setup in this case is easier 😅).

By the way, this LGTM!

**Comment by leonardost** [comment]
@gius80 thanks for the review! That's a good point, maybe we can add some text explaining that if you can't add NS records, you should use the advanced setup.

And I agree that the "advanced" one is simpler 😆 the thing is the suggested setup with NS records should be more robust to change - whenever a site is converted to/from atomic, the site's IP and WPCOM domain change, so CNAME and A records would need to be updated, whereas NS records wouldn't.

**Comment by gius80** [comment]
> maybe we can add some text explaining that if you can't add NS records, you should use the advanced setup.

That would be great, thanks!

> And I agree that the "advanced" one is simpler 😆 the thing is the suggested setup with NS records should be more robust to change - whenever a site is converted to/from atomic, the site's IP and WPCOM domain change, so CNAME and A records would need to be updated, whereas NS records wouldn't.

Gotcha, I didn't consider this :)

**Comment by delputnam** [comment]
> My only doubt is about the Suggested Setup, not all providers allow you to create NS [for subdomains]

This^ is exactly my worry too.  I wonder if we should address this issue directly. One thought I had was to change the "Switch to our advanced setup" text to something like "Can't set NS records on your subdomain? Switch to our advanced setup."

I have also sort of wondered if we should change the name of "Advanced setup" to "Alternative setup" or something that makes it sound less "scary". Especially since as @gius80 mentions, it actually seems less advanced in this case. :-D

**Comment by leonardost** [comment]
@delputnam  I like these suggestions 👍 I'll add a "Can't set NS records on your subdomain? Switch to our alternative setup" text there.

I also think "alternative" makes more sense here 😄 there was this same discussion in the old mapping component I touched a year ago 😂 https://github.com/Automattic/wp-calypso/pull/51251#pullrequestreview-622990589

So I guess it makes sense to change "Advanced" to "Alternative" in the root domain connection setup too?

**Review by delputnam — APPROVED** [review]
This looks good to me.  There are still a number of other improvements we can make to try to help users get their mapped domains connected, but this is a good iteration for subdomains, I think.

Thanks for all the hard work on this one @leonardost !

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7156518

Thank you @leonardost for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7156518).

**Pull request merged**